pipeline/frontend/yaml/error.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2022 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yaml
+
+// PipelineParseError is an error that occurs when the pipeline parsing fails.
+type PipelineParseError struct {
+	Err error
+}
+
+func (e PipelineParseError) Error() string {
+	return e.Err.Error()
+}
```

Comment:
```suggestion
}

// IsPipelineParseError checks if an error is a PipelineParseError
func IsPipelineParseError(err error) bool {
	_, ok1 := err.(PipelineParseError)
	_, ok2 := err.(*PipelineParseError)
	return ok1 || ok2
}
```

Reply:
```go
// IsPipelineParseError checks if an error is a PipelineParseError
func IsPipelineParseError(err error) bool {
	_, ok := err.(*PipelineParseError)
	return ok
}
```
would do it too ... if we are sure we only use &PipelineParseError{} version ...

Reply:
Why not just use errors.As? As that would work also for wrapped errors

Reply:
https://pkg.go.dev/errors#As

Reply:
https://go.dev/play/p/pT0g50DWTku vs https://go.dev/play/p/2X1yEErc5XO I would say ...

in the end if this is the only blocking thing I'm fine

Reply:
changed to `errors.Is`: https://go.dev/play/p/hpv3yhmlvzk